src/foraging_gui/Foraging.py

```diff
@@ -3625,6 +3684,7 @@ def exception_hook(self, exc_type, exc_value, exc_traceback):
     win.commit_ID=commit_ID
     win.current_branch=current_branch
     win.repo_url=repo_url
+    win.repo_dirty_flag=repo_dirty_flag
```

Comment:
Do you want to save the repo_dirty_flag to the json?

Reply:
Yes. This is critical information

Reply:
It seems it's not saved?

Reply:
Good catch, I've added it to the saved data